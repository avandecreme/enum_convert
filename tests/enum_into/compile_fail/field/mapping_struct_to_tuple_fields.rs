use enum_convert::EnumInto;

#[derive(EnumInto)]
#[enum_into(Target)]
enum Source {
    #[enum_into(Target::Tuple)]
    Struct {
        #[enum_into(Target::Tuple.0)]
        a: i32,
        #[enum_into(Target::Tuple.1)]
        b: i32,
    },
}

enum Target {
    Tuple(i32, i32),
}

fn main() {}
