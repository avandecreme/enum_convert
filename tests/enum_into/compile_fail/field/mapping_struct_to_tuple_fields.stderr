error: Unexpected mapping to positional field for struct variant
 --> tests/enum_into/compile_fail/field/mapping_struct_to_tuple_fields.rs:8:35
  |
8 |         #[enum_into(Target::Tuple.0)]
  |                                   ^
